src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentProperty.java

```diff
@@ -44,12 +48,16 @@
 	private final boolean isScore;
 	private final boolean isParent;
 	private final boolean isId;
+	private final boolean isAnnotatedFieldName;
```

Comment:
I would drop that field and do a null-check in `getFieldName`. `getAnnotatedFieldName` should return `null` in case the name is empty or blank.

Reply:
had it in there in order to not have a hasText() call in every getFieldName call. Switched to null check as you suggested